### PR TITLE
feat: add python3.11 support

### PR DESCRIPTION
**Pull request by moelasmar**
Adding the new python3.11 build image.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by hawflau — APPROVED** [review]

File: tests/test_build_images.py [hunk 1]
```diff
@@ -416,6 +416,37 @@ def test_packages(self):
         self.assertTrue(self.check_package_output("python --version", "Python 3.10."))
         self.assertTrue(self.is_package_present("pip"))
 
+@pytest.mark.python311
+class TestBIPython311ForArm(BuildImageBase):
+    __test__ = True
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass("python3.11", "Dockerfile-python310", "pip", tag="arm64")
```
Comment (mndeveci) [inline tests/test_build_images.py]: Apologies not seeing this earlier but this seems like using py3.10 dockerfile?

Comment (moelasmar) [reply tests/test_build_images.py]: fixed

**Review by mndeveci — APPROVED** [review]

File: build-image-src/Dockerfile-python311 [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+# Install AWS CLI
+ARG AWS_CLI_ARCH
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-$AWS_CLI_ARCH.zip" -o "awscliv2.zip" && unzip awscliv2.zip && ./aws/install && rm awscliv2.zip && rm -rf ./aws
+
+# Install SAM CLI in a dedicated Python virtualenv
+ARG SAM_CLI_VERSION
+RUN curl -L "https://github.com/awslabs/aws-sam-cli/archive/v$SAM_CLI_VERSION.zip" -o "samcli.zip" && \
+  unzip samcli.zip && python3 -m venv /usr/local/opt/sam-cli && \
+  /usr/local/opt/sam-cli/bin/pip3 --no-cache-dir install -r ./aws-sam-cli-$SAM_CLI_VERSION/requirements/base.txt && \
```
Comment (sriram-mv) [inline build-image-src/Dockerfile-python311:35]: Related PRs from: https://github.com/aws/aws-sam-cli/pull/5494

```
pip install wheel "cython<3.0.0" pyyaml==5.4.1 --no-build-isolation
pip uninstall wheel cython -y
```

also do we need no-cache-dir?

Comment (sriram-mv) [reply build-image-src/Dockerfile-python311:35]: Need to make changes to 3.10 as well. Pyyaml does not have wheels for 3.10/3.11

Comment (moelasmar) [reply build-image-src/Dockerfile-python311:35]: let me update it

Comment (moelasmar) [reply build-image-src/Dockerfile-python311:35]: done

File: build-image-src/Dockerfile-python311 [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-$AWS_CLI_ARCH.zip" -o "awscliv2.zip" && unzip awscliv2.zip && ./aws/install && rm awscliv2.zip && rm -rf ./aws
+
+# Install SAM CLI in a dedicated Python virtualenv
+# Note: Remove cython<3.0.0 line once the issue with building pyyaml from source is fixed
+ARG SAM_CLI_VERSION
+RUN curl -L "https://github.com/awslabs/aws-sam-cli/archive/v$SAM_CLI_VERSION.zip" -o "samcli.zip" && \
+  unzip samcli.zip && python3 -m venv /usr/local/opt/sam-cli && \
+  /usr/local/opt/sam-cli/bin/pip3 install wheel && \
+  /usr/local/opt/sam-cli/bin/pip3 install "cython<3.0.0" pyyaml==5.4.1 --no-build-isolation && \
+  /usr/local/opt/sam-cli/bin/pip3 uninstall wheel cython --yes && \
```
Comment (mndeveci) [inline build-image-src/Dockerfile-python311]: nit: can we move this line up so that cleanup will be removal of single line?

Comment (moelasmar) [reply build-image-src/Dockerfile-python311]: done

**Pull request merged**